core/include/traccc/geometry/silicon_detector_description.hpp

```diff
@@ -0,0 +1,240 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Local include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/definitions/qualifiers.hpp"
+
+// Detray include(s).
+#include <detray/geometry/barcode.hpp>
+
+// VecMem include(s).
+#include <vecmem/edm/container.hpp>
+
+namespace traccc {
+
+/// Interface for the @c traccc::silicon_detector_description class.
+///
+/// It provides the API that users would interact with, while using the
+/// columns/arrays defined in @c traccc::silicon_detector_description.
+///
+template <typename BASE>
+class silicon_detector_description_interface : public BASE {
+
+    public:
+    /// Inherit the base class's constructor(s)
+    using BASE::BASE;
+
+    /// @name Detector geometry information
+    /// @{
+
+    /// The identifier of the detector module's surface (non-const)
+    ///
+    /// Can be used to look up the module in a @c detray::detector object.
+    ///
+    /// @return A (non-const) vector of @c detray::geometry::barcode objects
+    ///
+    TRACCC_HOST_DEVICE
+    auto& surface_link() { return BASE::template get<0>(); }
```

Comment:
Renamed to `geometry_id`. This makes this a little inconsistent with the current `traccc::cell` and `traccc::measurement` classes. But since those will be rewritten soon as well, I'll just make them adhere to this "new" naming convention during the rewrite.